source/source.go

```diff
@@ -73,6 +74,21 @@ func getTTLFromAnnotations(annotations map[string]string) (endpoint.TTL, error)
 	return endpoint.TTL(ttlValue), nil
 }
 
+// parseTTL parses TTL from string, returning duration in seconds.
+// parseTTL supports both integers like "600" and durations based
+// on Go Duration like "10m", hence "600" and "10m" represent the same value.
+//
+// Note: for durations like "1.5s" the fraction is omitted (resulting in 1 second
+// for the example).
+func parseTTL(s string) (ttlSeconds int64, err error) {
+	ttlDuration, err := time.ParseDuration(s)
+	if err == nil {
```

Comment:
Can you do it the other way around? I think this is more go idiomatic.

Reply:
Sure, in this case it should be the other way around, I'll change this.

Reply:
Done